app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,22 @@
+module SessionsHelper
+  # Logs in the given user.
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  # Returns the current logged-in user (if any).
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
```

Comment:
giải thích cú pháp chỗ này

Reply:
> giải thích cú pháp chỗ này

Cái này đầu tiên là ||= nó giống như toán tử +=,ví dụ x+=1 ---> x=x+1.
Như vậy dòng 9 nó được hiểu là :
@current_user=@current_user || User.find_by(id: session[:user_id]) if session[:user_id].
Còn dòng này thì có hiểu nó ở 1 khía cạnh nào đó thì giống toán tử 3 ngôi,trả về kết quả là @current_user nếu nó khác nil,ngược lại trả về User.find_by(id: session[:user_id]) if session[:user_id].